### PR TITLE
apply on companion objects should return the companion class

### PR DESCRIPTION
**Pull request by rxin**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -370,8 +370,17 @@ arrayBuffer += elem
 
 ### <a name='apply_method'>apply Method</a>
 
-Avoid defining apply methods on classes. These methods tend to make the code less readable, especially for people less familiar with Scala. It is also harder for IDEs (or grep) to trace. In the worst case, it can also affect correctness of the code in surprising ways, as demonstrated in [Parentheses](#parentheses). It is however ok to define them in companion objects as factory methods.
+Avoid defining apply methods on classes. These methods tend to make the code less readable, especially for people less familiar with Scala. It is also harder for IDEs (or grep) to trace. In the worst case, it can also affect correctness of the code in surprising ways, as demonstrated in [Parentheses](#parentheses).
 
+A common pattern is to define apply methods on companion objects as factory methods. In these cases, the apply method should return the companion class type.
```
Comment (aarondav) [inline README.md]: Maybe rephrase: It is acceptable to define apply methods on companion objects as factory methods

Comment (rxin) [inline README.md]: pr?

Comment (rxin) [inline README.md]: nvm i'm going to fix it.
